File: BackEnd/java/ZaloMini/src/main/java/com/devmam/zalomini/entities/ChatGroup.java
package com.devmam.zalomini.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.CreationTimestamp;

import java.time.Instant;
import java.util.Date;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity(name = "ChatGroup")
@Table(name = "chat_groups")
public class ChatGroup {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Lob
    @Column(name = "description")
    private String description;

    @Lob
    @Column(name = "avatar_url")
    private String avatarUrl;

    @ColumnDefault("'group'")
    @Lob
    @Column(name = "group_type")
    private String groupType;

    @ColumnDefault("200")
    @Column(name = "max_members")
    private Integer maxMembers;

    @ColumnDefault("1")
    @Column(name = "is_active")
    private Boolean isActive;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "created_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "updated_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date updatedAt;

}

File: BackEnd/Zalo_mini/Zalo_mini/Models/chat_group.cs
using System;
using System.Collections.Generic;

namespace Zalo_mini.Models
{
    public partial class chat_group
    {
        public chat_group()
        {
            conversations = new HashSet<conversation>();
            group_invitations = new HashSet<group_invitation>();
            group_members = new HashSet<group_member>();
        }

        public long id { get; set; }
        public string name { get; set; } = null!;
        public string? description { get; set; }
        public string? avatar_url { get; set; }
        public string? group_type { get; set; }
        public int? max_members { get; set; }
        public bool? is_active { get; set; }
        public long? created_by { get; set; }
        public DateTime? created_at { get; set; }
        public DateTime? updated_at { get; set; }

        public virtual user? created_byNavigation { get; set; }
        public virtual ICollection<conversation> conversations { get; set; }
        public virtual ICollection<group_invitation> group_invitations { get; set; }
        public virtual ICollection<group_member> group_members { get; set; }
    }
}
